update Rust and Go

bcressey opened the pull request:
**Issue number:**
Fixes #38, fixes #40


**Description of changes:**
Update Rust to 1.49.0. This required a few changes beyond the version bump: changing how custom targets are created; adding the host triplet to the targets list, which is no longer done implicitly; and dropping an unnecessary `strip` command.

Update Go to 1.15.6.


**Testing done:**
Built the four SDK combinations - aarch64 and x86_64 host, aarch64 and x86_64 target.

Built aws-k8s-1.18 variants with the four SDKs. Verified that they came up, joined the cluster, had no errors in the journal.


**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

Commit:
update Go to 1.15.6

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
update Rust to 1.49.0

Adapt the creation of our custom targets to upstream changes.

Add the host triple to the targets list, since this no longer happens
implicitly.

Remove the strip invocation. The LLVM library is statically linked
into librustc_driver, so the dynamic library is not installed.

Signed-off-by: Ben Cressey <bcressey@amazon.com>

Commit:
bump version to 0.15.0

Signed-off-by: Ben Cressey <bcressey@amazon.com>

tjkirch commented on configs/rust/config-aarch64.toml.in:
```diff
@@ -1,5 +1,6 @@
 [build]
```
```suggestion
# https://github.com/rust-lang/rust/blob/master/config.toml.example
[build]
```
Might be worth pointing to the example in these configs?

webern reviewed (approved):
They changed the directory structure in the source? Is that why those paths changed?

zmrow reviewed (approved):
:microscope:

bcressey commented:
> They changed the directory structure in the source? Is that why those paths changed?

Correct, and there's a new scheme where some of the architecture-specific targets copy settings from a base set of options --
```
linux_base.rs -> linux_musl_base.rs -> x86_64_unknown_linux_musl.rs
```

The "unknown" vendor is the overall default for the options struct, so we have to apply it somewhere, and it was easiest to do that immediately after loading the base settings.

Commit:
add link to example Rust config

The example config is well-documented and provides context on what
these settings mean, along with the others that we leave at default
values.

Signed-off-by: Ben Cressey <bcressey@amazon.com>
